### DISCUSSION TITLE
命令深度

### DISCUSSION THREAD

**Discussion by RF-Tar-Railt** [ArcletProject/Alconna#44] [Ideas]
众所周知，Alconna 内包括 Subcommand和 Option 两大组件，Option与Subcommand的结合允许 Alconna 拓展其命令层数

```python
Alconna(
    "test", 
    options=[
        Subcommand("sub", options=[Option("opt")]
        )
    ]
)
```

其允许 `test sub opt <...>` 三层层数

但假设以下命令
`/test bot shedule add <...>`
其命令层数超过了 3层，Alconna 目前无法有效应对这种情况。

目前我预想了两种解决方案：
1. 允许Subcommand 嵌套 Subcommand

该方案最易实现，但容易造成因为嵌套过多导致的“杀人书”

2. Option 新增参数 require

例如 Option(bar, require=foo)

3. 引入新组件 Sentence

**Comment by GreyElaina** [comment]
直接做成 *match_sentences 不就好了。。

**Reply by RF-Tar-Railt** [reply]
那也需要确定在哪里传入sentence（
我倾向塞在Option里
